pkg/generate/code/compare.go

```diff
@@ -564,7 +564,8 @@ func CompareStruct(
 		var compareConfig *ackgenconfig.CompareFieldConfig
 		// memberFieldPath contains the field path along with the prefix cfg.PrefixConfig.SpecField + "." hence we
 		// would need to substring to exclude cfg.PrefixConfig.SpecField + "." to get correct field config.
-		fieldConfig := fieldConfigs[memberFieldPath[len(cfg.PrefixConfig.SpecField) + 1 :len(memberFieldPath)]]
+		specFieldLen := len(strings.TrimPrefix(cfg.PrefixConfig.SpecField, "."))
```

Comment:
Can we have this change in a different PR? I know it's a small change but it will be messy if we ever have to revert this commit.

Reply:
Done